manta/src/main/java/ch/cyberduck/core/manta/MantaDirectoryFeature.java

```diff
@@ -0,0 +1,66 @@
+package ch.cyberduck.core.manta;
+
+/*
+ * Copyright (c) 2002-2017 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.DefaultIOExceptionMappingService;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.features.Directory;
+import ch.cyberduck.core.features.Write;
+import ch.cyberduck.core.transfer.TransferStatus;
+
+import java.io.IOException;
+
+import com.joyent.manta.exception.MantaException;
+
+public class MantaDirectoryFeature implements Directory {
+
+    private final MantaSession session;
+
+    public MantaDirectoryFeature(MantaSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public Path mkdir(final Path directory, final String region, final TransferStatus status) throws BackgroundException {
+        try {
+            session.getClient().putDirectory(session.pathMapper.requestPath(directory));
+        }
+        catch(MantaException e) {
+            throw session.exceptionMapper.map("Cannot create folder {0}", e, directory);
+        }
+        catch(IOException e) {
+            throw new DefaultIOExceptionMappingService().map("Cannot create folder {0}", e, directory);
+        }
+        return directory;
+    }
+
+    @Override
+    public boolean isSupported(final Path workdir) {
+        return session.pathMapper.isUserWritable(workdir);
+    }
+
+    /**
+     * TODO: what does this do?
```

Comment:
Empty implementation is just fine because this feature does not make use of `Write` interface.